### ISSUE TITLE
Paws::Exception 'message' attribute should be nullable

### ISSUE THREAD

**Issue by karenetheridge** [pplu/aws-sdk-perl#194]
It seems that exception messages can sometimes be 'null', but the attribute type constraint doesn't allow it.  e.g. I'm seeing this in my logs:

```

2017-09-27 15:33:39-0700 - Attribute (message) does not pass the type constraint because: Validation failed for '__ANON__' with value undef at .../lib/perl5/x86_64-linux-gnu-thread-multi/Moose/Meta/Class.pm line 275
Moose::Meta::Class::new_object('Moose::Meta::Class=HASH(0xdeadbeef)', 'HASH(0xdeadbeef)') called at (eval 29999) line 20
Throwable::Error::new('Paws::Exception', 'message', undef, 'code', 'ServiceUnavailableException', 'request_id', 'sha elided', 'http_status', 503) called at .../lib/perl5/Paws/Net/RestJsonResponse.pm line 72
Paws::Net::RestJsonResponse::error_to_exception('Paws::Lambda=HASH(0xdeadbeef)', 'Paws::Lambda::Invoke=HASH(0xdeadbeef)', 503, '{"Message":null}', 'HASH(0xdeadbeef)') called at .../lib/perl5/Paws/Net/RestJsonResponse.pm line 20
Paws::Net::RestJsonResponse::handle_response('Paws::Lambda=HASH(0xdeadbeef)', 'Paws::Lambda::Invoke=HASH(0xdeadbeef)', 503, '{"Message":null}', 'HASH(0xdeadbeef)') called at .../lib/perl5/Paws/Net/Caller.pm line 40
Paws::Net::Caller::caller_to_response('Paws::Net::Caller=HASH(0xdeadbeef)', 'Paws::Lambda=HASH(0xdeadbeef)', 'Paws::Lambda::Invoke=HASH(0xdeadbeef)', 503, '{"Message":null}', 'HASH(0xdeadbeef)') called at .../lib/perl5/Paws/Net/RetryCallerRole.pm line 19
Paws::Net::RetryCallerRole::do_call('Paws::Net::Caller=HASH(0xdeadbeef)', 'Paws::Lambda=HASH(0xdeadbeef)', 'Paws::Lambda::Invoke=HASH(0xdeadbeef)') called at .../lib/perl5/Paws/Lambda.pm line 84
...
```

(This is under Paws version 0.33.)

**Comment by karenetheridge** [comment]
somewhat related, I also see this sometimes:

`Use of uninitialized value $code in pattern match (m//) at .../lib/perl5/Paws/Net/JsonResponse.pm line 62.`

**Comment by pplu** [comment]
Hi,

  Thanks for the report. I think it will be solved in the next release (0.35) or maybe 0.36 (depending on when I can get to merging all the refactors together), as I've refactored the handling of responses to behave a bit more forgiving with the errors. I'll try to incorporate a test (if I havent already) to make sure this type of response is handled gracefully.

**Comment by pplu** [comment]
Paws 0.37 with this change has just been uploaded to CPAN
